### PR TITLE
ADEN-4124 Video in BFAA

### PR DESCRIPTION
**Pull request by drets**
• expand video on click
• hide video after playing

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [bfaa.js#L16](https://github.com/Wikia/app/blob/68af69e274baa86035b02921216ce091195defb7/extensions/wikia/AdEngine/js/template/bfaa.js#L16): Function has 13 parameters which is greater than 7 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [bfaa.js#L14](https://github.com/Wikia/app/blob/235a72fe82ae381990654d1e32e918aed89fb777/extensions/wikia/AdEngine/js/template/bfaa.js#L14): Function has 11 parameters which is greater than 7 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by bognix** [comment]
I am wondering if you considered adding another template instead of re-using the BFAA template. I understand why you choose re-usage and I'm asking just out of the curiosity.

My second question is, have you considered breaking BFAA template into two: bfaa_desktop and bfaa_mobile. Logic for loading BFAA on those two platforms is totally different and we kind of do this by having mobileHandler and desktopHandler. How about going one step further and having two files `bfaaOasis` and `bfaaMercury`, then depending on the platform user is on we require one or the other inside `bfaa.js` file

Third comment, I tried to run this code on my devbox and I had this error in console:
`GET http://redirector.googlevideo.com/videoplayback?id=8de36535d23ae926&itag=22… 403 (Forbidden)`
And after clicking on image I had this error:
`uapVideoAd.js:37 Uncaught (in promise) DOMException: The element has no supported sources.`

**Review by bognix — CHANGES_REQUESTED** [review]
Please address my third question/comment in the comment to PR

File: extensions/wikia/AdEngine/js/template/bfaaDesktop.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	'ext.wikia.adEngine.adHelper',
+	'ext.wikia.adEngine.context.uapContext',
+	'ext.wikia.adEngine.provider.btfBlocker',
+	'ext.wikia.adEngine.slotTweaker',
+	'ext.wikia.adEngine.video.uapVideoAd',
+	'wikia.document',
+	'wikia.log',
+	'wikia.window',
+	require.optional('ext.wikia.aRecoveryEngine.recovery.tweaker')
+], function (
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/template/bfaaDesktop.js:12]: MAJOR Function has 9 parameters which is greater than 7 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)

File: extensions/wikia/AdEngine/js/template/bfaaMobile.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+define('ext.wikia.adEngine.template.bfaaMobile', [
+	'ext.wikia.adEngine.adHelper',
+	'ext.wikia.adEngine.context.uapContext',
+	'ext.wikia.adEngine.provider.btfBlocker',
+	'ext.wikia.adEngine.slotTweaker',
+	'wikia.document',
+	'wikia.log',
+	'wikia.window',
+	require.optional('ext.wikia.adEngine.mobile.mercuryListener')
+], function (
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/template/bfaaMobile.js:11]: MAJOR Function has 8 parameters which is greater than 7 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
- MAJOR 2 major

Watch the comments in this conversation to review them.

File: skins/oasis/css/core/ads.scss [hunk 3]
```diff
@@ -81,6 +81,11 @@ $taboola-small-breakpoint: "#{$screen} and (max-width: 420px)";
 			}
 		}
 	}
+
+	video {
```
Comment (bognix) [inline skins/oasis/css/core/ads.scss]: do you think it's safe to style all videos inside top leaderboard?

**Review by bognix — APPROVED** [review]

**Pull request merged**